vespa-athenz/src/main/java/com/yahoo/vespa/athenz/client/zts/ZtsClient.java

```diff
@@ -36,4 +40,53 @@ InstanceIdentity refreshInstance(AthenzService providerIdentity,
                                      String instanceId,
                                      boolean requestServiceToken,
                                      Pkcs10Csr csr);
+
+    /**
+     * Fetch a role token for the target domain
+     *
+     * @param domain Target domain
+     * @return A role token
+     */
+    ZToken getRoleToken(AthenzDomain domain);
+
+    /**
+     * Fetch a role token for the target domain and role
+     *
+     * @param domain Target domain
+     * @param roleName Target role
```

Comment:
there's nothing in here in the API indicating roleName can be null? Unless nullability can be inferred from e.g. the type, one has to be a bit more careful about documentation IMO